torch/csrc/jit/ir/ir.h

```diff
@@ -1512,5 +1518,68 @@ struct OperatorSet {
   std::unordered_map<Symbol, std::vector<std::shared_ptr<Operator>>> ops;
 };
 
+template <typename T>
+struct OperatorMap {
+  OperatorMap() {}
+  OperatorMap(std::initializer_list<std::pair<std::string, T>> init) {
+    insert(init);
+  }
+
+  bool contains(const Symbol& key) const {
+    return map.count(key) != 0;
+  }
+
+  bool contains(const std::string& key) const {
+    return map.count(Symbol::fromQualString(key)) != 0;
+  }
+
+  void insert(const Symbol& key, T value) {
+    map[key] = value;
+  }
+
+  void insert(const std::string& key, T value) {
+    map[Symbol::fromQualString(key)] = value;
+  }
+
+  void insert(std::initializer_list<std::pair<Symbol, T>> v) {
+    for (auto el : v) {
+      map[el.first] = el.second;
+    }
+  }
+
+  void insert(std::initializer_list<std::pair<std::string, T>> v) {
+    for (auto el : v) {
+      map[Symbol::fromQualString(el.first)] = el.second;
+    }
+  }
+
+  void erase(const Symbol& key) {
+    map.erase(key);
+  }
+
+  void erase(const std::string& key) {
+    map.erase(Symbol::fromQualString(key));
+  }
+
+  T find_or(const Symbol& key, T t = T()) {
+    auto it = map.find(key);
+    if (it != map.end()) {
+      return it->second;
+    }
+    return t;
+  }
+
+  c10::optional<T> find(const Symbol& key) {
+    auto it = map.find(key);
+    if (it != map.end()) {
+      return it->second;
+    }
+    return c10::nullopt;
+  }
+
+ private:
+  std::unordered_map<Symbol, T> map;
```

Comment:
In `OperatorSet` and `OperatorMap` we will have multiple schemas with the same node kind. For example:
```
      "aten::mul.Tensor(Tensor self, Tensor other) -> Tensor",
      "aten::mul.Scalar(Tensor self, Scalar other) -> Tensor",
```

I think, as in OperatorSet, we want to map from Symbol to a vector, and then within the vector store pairs of `std::shared_ptr<Operator>, T`